Implemented Some of the TODO features and added one I wanted

Technical-1 opened the pull request:
# Implemented Features:
1. Date Input Support
- Detects date fields using the input[type='date'] selector
- Handles conversion from common date formats (DD/MM/YYYY and MM/DD/YYYY) to the HTML standard YYYY-MM-DD

2. Adaptive Levenshtein Distance Threshold - Replaced the static threshold (79) with an adaptive function
 - Adjusts the required similarity threshold based on string length
- Uses stricter matching (90%) for very short strings where small differences matter more
- Gradually relaxes the threshold for longer strings (down to 75%)

3. CSV Import Functionality - Added the ability to import form data from CSV files
- Users can now load prepared form data from CSV files

Technical-1 committed:
Implemented date input

Technical-1 committed:
Added csv import of keys and values

Technical-1 committed:
Removed time as not working

muratalperen commented:
![image](https://github.com/user-attachments/assets/d213fe2d-3526-471c-9e39-2f7b559da065)
Hi, really thanks for the features! Have you checked the date input support? How does it work? Can you give me an example of it working, I couldn't get it to work.

Technical-1 commented:
<img width="1715" alt="image" src="https://github.com/user-attachments/assets/b8c13f5c-9222-4642-b0bb-7fe87d011545" />

Technical-1 commented:
When I tested it while developing it worked immediately. Could it possibly be the locale?
Here is the form link I am using to test with https://docs.google.com/forms/d/1I-QOMiwzEgOUBdCZHTR3Z9STLv-aoSox5z34zd1jUx8/viewform

jjromero2927-hue commented:
Hi, is there some video to step by step process?

NEED HELP...

Technical-1 committed:
foundation: shared storage helper, storage.local backend, sync->local migration

Add scripts/storage.js exposing GFAFStorage: getFormData/setFormData,
getSettings/setSettings, and an idempotent migrateSyncToLocal. All writes
check chrome.runtime.lastError so failures are surfaced instead of silently
lost. Move the backend from chrome.storage.sync to chrome.storage.local,
removing the sync per-item/total quota risk.

- manifest content_scripts loads storage.js before GoogleForm.js
- popup loads storage.js before menu.js; reads/writes go through the helper
- content script and popup run a one-time, marker-guarded migration that
  never clobbers existing local data, bailing out on a local read error
- CSV import save path routed through the helper so write errors surface

Includes IMPLEMENTATION_PLAN.md documenting all four buckets.

Technical-1 committed:
Merge pull request #1 from Technical-1/foundation/storage-local-helper

Foundation: shared storage helper + storage.local backend + migration

Technical-1 committed:
fill: observer-safe filling, configurable date locale, match caching

A1: hoist the MutationObserver to module scope, disconnect it during the
programmatic write pass (reconnect in finally), and debounce observer-driven
fills by 120ms so dispatched input/change events can no longer re-trigger an
endless re-fill loop.

A2: guard CalculateSimilarity against a zero max length (return 100) and skip
empty form titles.

A3: replace the unreachable duplicate-regex date branch with a pure
toISO(answer, fmt) and a settings-aware handler fill(el, answer, ctx). A new
popup date-format selector (Auto / DD-MM-YYYY / MM-DD-YYYY) persists via the
storage helper; the content script reads it on every fill.

A4: memoize title->match keyed on the formData key set and clear it on
storage.onChanged so popup edits re-fill live.

Technical-1 committed:
Merge pull request #2 from Technical-1/fill-engine

Fill Engine: observer loop fix, date locale, NaN guard, match cache

Technical-1 committed:
csv: shared quote/newline-safe parser module; remove dead duplicate

Extract parseCSV/processCSVFile into scripts/csvParser.js (GFAFCsv namespace),
loaded by the popup after storage.js. Replace the line-split+regex parser with
a single-pass state machine that correctly handles commas and newlines inside
quoted fields, "" escapes, a leading BOM, and CRLF/CR/LF endings; an
unterminated quote at EOF is flushed instead of throwing.

processCSVFile now persists through GFAFStorage, aborts the import (instead of
wiping existing entries) if the read fails, and reports a write failure rather
than a false success. Delete the dead, never-loaded scripts/csvImporter.js.

Technical-1 committed:
Merge pull request #3 from Technical-1/csv-import

CSV Import: shared quote/newline-safe parser module

Technical-1 committed:
background: real save-on-close port flush; remove stray debug log

Replace the no-op service-worker listener with a save-on-close handler: the
popup opens a popupSession port and streams the latest form data on every
SaveData; the worker buffers it and flushes to chrome.storage.local on port
disconnect, covering the case where the popup closes before its own async
write settles. Register background.js as the MV3 service_worker so the
listener actually runs. Remove the stray console.log("formData") in
DisplayData.

Technical-1 committed:
Merge pull request #4 from Technical-1/background-cleanup

Background: real save-on-close port flush + cleanup

Technical-1 committed:
chore: remove internal planning artifacts

Technical-1 committed:
docs: add portfolio docs and refresh README

Technical-1 committed:
Merge pull request #5 from Technical-1/docs-cleanup

Docs cleanup: remove planning artifacts, add portfolio docs

Technical-1 committed:
matcher: token-coverage matching + dependency-free core tests

Field titles now match short saved keys two ways — whole-string edit
distance (unchanged) and per-word token coverage — so verbose titles
like "Email Address" / "Phone Number" fill from a saved "Email" / "Phone".
Previously only small whole-string differences matched, so the README's
"Email Address maps to the same answer" claim was not actually true.

Token coverage only raises a score (it returns 0 when any token is
uncovered), so it adds matches without weakening precision: "Username"
still does not match "Name".

Add test/core.test.js, a zero-dependency Node harness that loads the real
sources in a vm sandbox (stubbing browser globals) and covers the CSV
parser, date normalization, and field matching — 28 assertions. Update
README/qa/Changelog to describe the two-path matcher accurately.

Technical-1 committed:
Merge pull request #6 from Technical-1/matcher/token-coverage

matcher: token-coverage matching + dependency-free core tests

Technical-1 committed:
docs: describe two-path matcher and the new test harness

Update the portfolio architecture and stack docs to match the current
code: the matcher now scores titles by whole-string Levenshtein OR
per-word token coverage (whichever is higher), and there is now a
dependency-free Node test harness covering the CSV parser, date
normalization, and field matching.

Technical-1 committed:
Merge pull request #7 from Technical-1/docs/matcher-testing-refresh

docs: describe two-path matcher and the new test harness

Technical-1 committed:
assets: add portfolio OG image and reusable template

Add .portfolio/og-image.png (1200x630 @2x), a social preview / marketing
image for the project. The composition uses the extension's purple I-beam
brand mark and a stylized auto-filling form card that surfaces the fuzzy
field matching (Email -> Email Address).

Include .portfolio/og-previews/og-image.html, the self-contained template
(brand icon inlined as base64) used to render the PNG, so the image can be
regenerated with a headless-Chrome screenshot.

Technical-1 committed:
Merge pull request #8 from Technical-1/assets/og-image

assets: add portfolio OG image and reusable template

Technical-1 committed:
fixed name
